File: data_structures_by_c++/arrays/array_insertion.cpp
#include<bits/stdc++.h>
using namespace std;

#define int long long

int32_t main()
{
    // ios::sync_with_stdio(false);
    // cin.tie(nullptr);

    int n;
    cout << "Enter number of elements: ";
    cin >> n;

    int arr[100];

    cout << "Enter elements:" << endl;
    for (int i = 0; i < n; i++)
    {
        cin >> arr[i];
    }

    int pos, val;
    cout << "Enter position to insert (0-based index): ";
    cin >> pos;

    cout << "Enter value to insert: ";
    cin >> val;

    for (int i = n; i > pos; i--)
    {
        arr[i] = arr[i - 1];
    }

    arr[pos] = val;
    n++;

    cout << "Array after insertion: ";
    for (int i = 0; i < n; i++)
    {
        cout << arr[i] << " ";
    }
}

File: data_structures_by_python/arrays/array_insertion.py
# arr = list(map(int, input("Enter elements: ").split()))
# pos = int(input("Enter position: "))
# val = int(input("Enter value: "))

# arr.insert(pos, val)

# print("After insertion: ", arr)

## without stl
n = int(input("Enter number of elements: "))
arr = [0]*100
print("Enter element: ")
for i in range(n):
    arr[i] = int(input())
pos = int(input("Enter position to insert (0 based): "))
val = int(input("Enter value to insert: "))

i = n
while i > pos:
    arr[i] = arr[i - 1]
    i -= 1
    
arr[pos] = val
n += 1
print("After insertion: ", end=" ")
for i in range(n):
    print(arr[i], end=" ")